src/feature/control/control_getinfo.c

```diff
@@ -325,47 +325,37 @@ getinfo_helper_current_time(control_connection_t *control_conn,
   return 0;
 }
 
-/**
- * Switch between microdesc vs networkstatus descriptor dumps
- * Assumes that question would be either:
- * - dir/status-vote/current/consensus-microdesc
- * or
- * - dir/status-vote/current/consensus
- */
+/** GETINFO helper for dumping different consensus flavors
+ * returns: 0 on success -1 on error. */
 STATIC int
-getinfo_helper_current_consensus(const char* question,
-                          char** answer,
-                          const char** errmsg)
+getinfo_helper_current_consensus(int flavor,
+                                 char** answer,
+                                 const char** errmsg)
 {
-  // Ensures question contains a request for
-  // ns or microdesc consensus
-  if (
-      strcmp(question, "dir/status-vote/current/consensus") &&
-      strcmp(question, "dir/status-vote/current/consensus-microdesc")) {
-    return 0;
+  const char *flavor_name = networkstatus_get_flavor_name(flavor);
+  if (!strcmp(flavor_name, "")) {
```

Comment:
This strcmp with "" is no longer useful after using `networkstatus_get_flavor_name`.  At least change it to check for "??" or change the general check logic.